ors-engine/src/main/java/org/heigit/ors/snapping/SnappingRequest.java

```diff
@@ -0,0 +1,49 @@
+package org.heigit.ors.snapping;
+
+import com.graphhopper.GraphHopper;
+import com.graphhopper.routing.util.AccessFilter;
+import com.graphhopper.routing.util.FlagEncoder;
+import com.graphhopper.storage.RoutingCHGraph;
+import com.graphhopper.util.PMap;
+import org.heigit.ors.common.ServiceRequest;
+import org.heigit.ors.matrix.MatrixSearchContext;
+import org.heigit.ors.matrix.MatrixSearchContextBuilder;
+import org.heigit.ors.routing.RoutingProfileType;
+import org.heigit.ors.routing.WeightingMethod;
+import org.heigit.ors.util.ProfileTools;
+import org.locationtech.jts.geom.Coordinate;
+
+public class SnappingRequest extends ServiceRequest {
+    private int profileType;
+    private Coordinate[] locations;
+    private double maximumSearchRadius;
+
+    public SnappingRequest(int profileType, Coordinate[] locations, double maximumSearchRadius) {
+        this.profileType = profileType;
+        this.locations = locations;
+        this.maximumSearchRadius = maximumSearchRadius;
+    }
+
+    public SnappingResult computeResult(GraphHopper gh) throws Exception {
+        String encoderName = RoutingProfileType.getEncoderName(profileType);
+        FlagEncoder flagEncoder = gh.getEncodingManager().getEncoder(encoderName);
+        PMap hintsMap = new PMap();
+        int weightingMethod = WeightingMethod.RECOMMENDED; // Only needed to create the profile string
+        ProfileTools.setWeightingMethod(hintsMap, weightingMethod, profileType, false);
+        ProfileTools.setWeighting(hintsMap, weightingMethod, profileType, false);
+        String CHProfileName = ProfileTools.makeProfileName(encoderName, hintsMap.getString("weighting", ""), false);
+        String profileName = CHProfileName;
+
+        // TODO: replace usage of matrix search context by snapping-specific class
```

Comment:
What does the to do depend on, could this already be part of this PR?

Reply:
It mainly depends on priorities. Implementing these snapping-specific classes is only worth, if we do it cleanly. This would require significant effort with moderate impact. The list of more important cleanup issues is long.

Reply:
At the time being the snap endpoint works *only if*  CH has been enabled for a given profile and fails otherwise 🙈 This happens because of the dependency on the `routingCHGraph` in the following call.

https://github.com/GIScience/openrouteservice/blob/b392a0c5681fe73afa1ae1559bb6452598fba772/ors-engine/src/main/java/org/heigit/ors/snapping/SnappingRequest.java#L41